src/libraries/System.Memory/tests/ReadOnlySpan/Count.T.cs

```diff
@@ -0,0 +1,402 @@
+            {
+                string[] a = new string[length];
+                for (int i = 0; i < length; i++)
+                {
+                    a[i] = (10 * (i + 1)).ToString();
+                }
+                ReadOnlySpan<string> span = new ReadOnlySpan<string>(a);
+
+                for (int targetIndex = 0; targetIndex < length - 1; targetIndex++)
+                {
+                    ReadOnlySpan<string> target = new string[] { a[targetIndex], a[targetIndex + 1] };
+
+                    int count = span.Count(target);
+                    Assert.Equal(1, count);
+                }
+            }
+        }
+
+        [Fact]
+        public static void TestNoMatchCount_String()
+        {
+            var rnd = new Random(42);
+            for (int length = 0; length <= byte.MaxValue; length++)
+            {
+                string[] a = new string[length];
+                string target = (rnd.Next(0, 256)).ToString();
+                for (int i = 0; i < length; i++)
+                {
+                    string val = (i + 1).ToString();
+                    a[i] = val == target ? (target + 1) : val;
+                }
+                ReadOnlySpan<string> span = new ReadOnlySpan<string>(a);
+
+                int count = span.Count(target);
+                Assert.Equal(0, count);
+            }
+        }
+        
+        [Fact]
+        public static void TestNoMatchCount_RosString()
+        {
+            var rnd = new Random(42);
+            for (int length = 0; length <= byte.MaxValue; length++)
+            {
+                string[] a = new string[length];
+                ReadOnlySpan<string> target = new string[] { rnd.Next(0, 256).ToString(), "0" };
+                for (int i = 0; i < length; i++)
+                {
+                    string val = (i + 1).ToString();
+                    a[i] = val == target[0] ? (target[0] + 1) : val;
+                }
+                ReadOnlySpan<string> span = new ReadOnlySpan<string>(a);
+
+                int count = span.Count(target);
+                Assert.Equal(0, count);
+            }
+        }
+
+        [Fact]
+        public static void TestMultipleMatchCount_String()
+        {
+            for (int length = 2; length < 32; length++)
+            {
+                string[] a = new string[length];
+                for (int i = 0; i < length; i++)
+                {
+                    a[i] = (10 * (i + 1)).ToString();
+                }
+
+                a[length - 1] = "5555";
+                a[length - 2] = "5555";
+
+                ReadOnlySpan<string> span = new ReadOnlySpan<string>(a);
+                int count = span.Count("5555");
+                Assert.Equal(2, count);
+            }
+        }
+
+        [Fact]
+        public static void TestMultipleMatchCount_RosString()
+        {
+            for (int length = 4; length < 32; length++)
+            {
+                string[] a = new string[length];
```

Comment:
It'd be good to have some string tests that validate:
1) It behaves like StringComparison.Ordinal, e.g. make sure things that would compare equivalent under a certain culture but not under ordinal don't compare equivalent, make sure things that would compare equivalent under OrdinalIgnoreCase but not under ordinal don't compare equivalent, etc.
2) That overlaps aren't counted, e.g. searching for "aa" in "aaaaa" should produce 2, not 4.

Reply:
Also, I didn't notice... are there any tests that exercise the path in the multiple values overload where it in turn just delegates to the single value overload?